app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php

```diff
@@ -127,11 +128,14 @@ public function execute()
             $attributeCode
         );
 
-        if ($attribute->getId() && !$attributeId || $attributeCode === 'product_type' || $attributeCode === 'type_id') {
-            $message = strlen($this->getRequest()->getParam('attribute_code'))
-                ? __('An attribute with this code already exists.')
-                : __('An attribute with the same code (%1) already exists.', $attributeCode);
-
+        if ($attribute->getId() && !$attributeId || in_array($attributeCode, static::RESERVED_CODES)) {
+            if (in_array($attributeCode, static::RESERVED_CODES)) {
```

Comment:
Let's simplify logic in this condition. The same `in_array` is being called twice. 
Please use `self` instead of `static` for constant access.